### DISCUSSION TITLE
Setting up NextAuth using Spotify Provider.

### DISCUSSION THREAD

**Discussion by KrychuTM** [nextauthjs/next-auth#1893] [Help] [answered]
I'm creating a personal project and I'm having a hard time understanding how Next Auth works with the Spotify API. To be more exact I'm having trouble understanding the use of the refresh token. Does Next Auth take care of that for us or do we have to set it up on our own? And if so is the refresh token necessary? I've already figured out how to log in and check if a user is logged in but just curious if that's all i need to do to start collecting data from the api.

**Comment by balazsorban44** [comment, marked as answer]
token rotation is currently not a built in feature, but we have a user land tutorial for it.

it might help you get started:
https://next-auth.js.org/tutorials/refresh-token-rotation

**Reply by KrychuTM** [reply]
Yeah thanks I just did some further research and found and came across it. It's an example for the Google Refresh Token process. I'm currently editing it to play along with the Spotify Api. Hopefully it works.

**Reply by balazsorban44** [reply]
both use the same spec under, so it should be fine

**Comment by KrychuTM** [comment]
Thanks I got it working. For some reason I'm not getting the email and profile pic but I''ll figure it out eventually. Appreciate the help.
